### PR TITLE
fix(workflow): Only update chart when query is finished

### PR DESCRIPTION
**Pull request by iProgramStuff**
This PR aims to fix a few known issues:
Fixed chart re-rendering too much as you type: (https://app.asana.com/0/1143070838064087/1163136329300643)
Fixed error message appearing as you type (https://app.asana.com/0/1143070838064087/1163355932236027) - the message will only appear when you 'apply' the filter via enter, etc.
Addresses https://app.asana.com/0/1143070838064087/1160225808505467 as far as I've seen so far, they work equivalently now

It accomplishes all this by not making the event-stats api call on text field change and keydown. That call is what triggered the error, re-rendering, and seemed to block tag filtering (i.e. on `stack.filename`)

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx [hunk 1]
```diff
@@ -409,7 +410,7 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
             onSubmit={this.handleSubmit}
             onSubmitSuccess={onSubmitSuccess}
             onCancel={this.handleCancel}
-            onFieldChange={this.handleFieldChange}
+            onFieldChange={debounce(this.handleFieldChange,200)}
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx]: The debounce is a good idea, but when I was poking it locally it felt like some of my changes didn't reflect in the chart until I refocused the query input and changed the text again.

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx [hunk 2]
```diff
@@ -43,7 +44,9 @@ class RuleConditionsForm extends React.PureComponent<Props, State> {
   state: State = {
     environments: null,
   };
-
+  blurFuncTest(...args){
+    console.log("blur test called",...args)
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx]: Some console.log left over.

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx [hunk 3]
```diff
@@ -123,9 +124,15 @@ class RuleConditionsForm extends React.PureComponent<Props, State> {
...
-                onBlur={onBlur}
                 onKeyDown={onKeyDown}
-                onSearch={query => onChange(query, {})}
+                onBlur={query => {
+                  onBlur(query);
+                  this.props.onFilterUpdate(query);
+                }}
+                onSearch={query => {
+                  this.props.onFilterUpdate(query);
+                  onChange(query, {});
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx]: Should the order that the onFilterUpdate and onChange/onBlur be the same?

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx [hunk 4]
```diff
@@ -123,9 +124,15 @@ class RuleConditionsForm extends React.PureComponent<Props, State> {
...
                 onChange={onChange}
-                onBlur={onBlur}
                 onKeyDown={onKeyDown}
-                onSearch={query => onChange(query, {})}
+                onBlur={query => {
+                  onBlur(query);
+                  this.props.onFilterUpdate(query);
+                }}
+                onSearch={query => {
+                  this.props.onFilterUpdate(query);
```
Comment (iProgramStuff) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx]: Does it have to be called in the reverse order as onBlur?

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx]: Order doesn't matter, but updated to be consistent

**Review by markstory — APPROVED** [review]
Looks good to me. Expanding the blur callback to receive the value lets you update the query and reload data when the conditions have changed.

**Pull request merged**